stage1/init/pod.go

```diff
@@ -517,37 +503,23 @@ func (p *Pod) appToNspawnArgs(ra *schema.RuntimeApp) ([]string, error) {
 		}
 	}
 
-	for _, mp := range app.MountPoints {
-		// there's already an injected mount for this target path, skip
-		if _, ok := mounts[mp.Path]; ok {
-			continue
-		}
-		vol, ok := vols[mp.Name]
-		if !ok {
-			catCmd := fmt.Sprintf("sudo rkt image cat-manifest --pretty-print %v", id)
-			volumeCmd := ""
-			for _, mp := range app.MountPoints {
-				volumeCmd += fmt.Sprintf("--volume %s,kind=host,source=/some/path ", mp.Name)
-			}
-
-			return nil, fmt.Errorf("no volume for mountpoint %q:%q in app %q.\n"+
-				"You can inspect the volumes with:\n\t%v\n"+
-				"App %q requires the following volumes:\n\t%v",
-				mp.Name, mp.Path, appName, catCmd, appName, volumeCmd)
-		}
-		ra.Mounts = append(ra.Mounts, schema.Mount{Volume: vol.Name, Path: mp.Path})
+	mounts, err := s1common.GenerateMounts(ra, vols)
+	if err != nil {
+		return nil, err
 	}
 
-	for _, m := range ra.Mounts {
+	for _, m := range mounts {
 		vol := vols[m.Volume]
 
 		opt := make([]string, 4)
 
 		// If the readonly flag in the pod manifest is not nil,
 		// then use it to override the readonly flag in the image manifest.
-		readOnly := isMPReadOnly(app.MountPoints, vol.Name)
+		var readOnly bool
```

Comment:
How about putting this snippet into `common`? Similar code is in two places now.